### PR TITLE
migrate more actions to protocol.xpack

### PR DESCRIPTION
**Pull request by talevy**
this PR creates client-side instances of the following actions:

- DeleteAction
- ForceMergeAction
- ReadOnlyAction
- RolloverAction
- ShrinkAction

AllocateAction can be found here (https://github.com/elastic/elasticsearch/pull/32853)

### CODE REVIEW

**Comment by talevy** [comment]
hey all. I've requested review. but I guess this review should be pending the review for Allocate Action since I want that one to govern how the rest of these are brought into the client

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java [hunk 1]
```diff
@@ -452,8 +452,24 @@ static Settings additionalSettings(final Settings settings, final boolean enable
...
+                    org.elasticsearch.protocol.xpack.indexlifecycle.ForceMergeAction::parse),
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.ReadOnlyAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.ReadOnlyAction.NAME),
+                    org.elasticsearch.protocol.xpack.indexlifecycle.ReadOnlyAction::parse),
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.RolloverAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.RolloverAction.NAME),
+                    org.elasticsearch.protocol.xpack.indexlifecycle.RolloverAction::parse),
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.ShrinkAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.ShrinkAction.NAME),
+                    org.elasticsearch.protocol.xpack.indexlifecycle.ShrinkAction::parse)
```
Comment (colings86) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java]: Same comment here as in the allocate action PR about needing a common `categoryClass` for all the actions so we can deserialise them without knowing the exact action type and also same comment about moving the  registration of the server side objects away from the client.

**Review by colings86 — CHANGES_REQUESTED** [review]
Same comments as the allocate action PR but I think this is good otherwise so we can apply the same fixes to both PRs 😄

File: x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/indexlifecycle/LifecycleAction.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.protocol.xpack.indexlifecycle;
+
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+
+/**
+ * Parent class for the client-side ilm actions for the {@link NamedXContentRegistry}
+ */
```
Comment (colings86) [inline x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/indexlifecycle/LifecycleAction.java]: Same comment as in http://github.com/elastic/elasticsearch/pull/32853

**Review by colings86 — APPROVED** [review]
LGTM

**Pull request merged**